File: SDKs/amazonaws.com/devops-guru/2020-12-01/go/pkg/models/shared/insighttimerange.go
package shared

import (
	"time"
)

// InsightTimeRange
//
//	A time ranged that specifies when the observed behavior in an insight started and ended.
type InsightTimeRange struct {
	EndTime   *time.Time `json:"EndTime,omitempty"`
	StartTime time.Time  `json:"StartTime"`
}

File: SDKs/amazonaws.com/devops-guru/2020-12-01/typescript/src/sdk/models/shared/insighttimerange.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// InsightTimeRange
/** 
 *  A time ranged that specifies when the observed behavior in an insight started and ended. 
**/
export class InsightTimeRange extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=EndTime" })
  endTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=StartTime" })
  startTime: Date;
}
